File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/ModulesViewModels/ReportsModuleViewModel.vb
Imports System
Imports DevExpress.DemoData.Helpers
Imports DevExpress.XtraReports

Namespace ProductsDemo.Modules

    Public Class ReportsModuleViewModel

        Shared Sub New()
            Call PatchDemoReportsConnectionStrings()
        End Sub

        Public Overridable Property DocumentSource As IReport

        Public Sub OnLoaded()
            UpdateReport()
        End Sub

        Public Sub OnUnloaded()
            DocumentSource.StopPageBuilding()
        End Sub

        Private Sub UpdateReport()
            DocumentSource = New DevExpress.DevAV.MasterDetailReport.Report()
            DocumentSource.CreateDocument(True)
        End Sub

        Private Shared Sub PatchDemoReportsConnectionStrings()
            Dim path As String = Utils.GetRelativePath("nwind.mdb")
            AppDomain.CurrentDomain.SetData("DataDirectory", IO.Path.GetDirectoryName(path))
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/ModulesViewModels/ReportsModuleViewModel.cs
using System;
using System.IO;
using System.Configuration;
using DevExpress.DemoData.Helpers;
using DevExpress.XtraReports;

namespace ProductsDemo.Modules {
    public class ReportsModuleViewModel {
        static ReportsModuleViewModel() {
            PatchDemoReportsConnectionStrings();
        }
        public virtual IReport DocumentSource { get; set; }

        public void OnLoaded() {
            UpdateReport();
        }
        public void OnUnloaded() {
            DocumentSource.StopPageBuilding();
        }
        void UpdateReport() {
            DocumentSource = new DevExpress.DevAV.MasterDetailReport.Report();
            DocumentSource.CreateDocument(true);
        }
        static void PatchDemoReportsConnectionStrings() {
            string path = Utils.GetRelativePath("nwind.mdb");
            AppDomain.CurrentDomain.SetData("DataDirectory", Path.GetDirectoryName(path));
        }
    }
}
